### DISCUSSION TITLE
request.page.name alternative to get the filepath inside of middleware

### DISCUSSION THREAD

**Discussion by vinaynair39** [vercel/next.js#38321] [Help]
Since` request.page `is now deprecated in v12.2.0, How can I get the file path of the requested page inside middleware.js?

**Comment by icyJoseph** [comment]
I don't think it is possible at this moment, at least for dynamic routes, every other route that maps directly to a file, I guess you could manually check. 

There's this other discussion where a hack was found: https://github.com/vercel/next.js/discussions/38159

I think that the hack depends on ordering and hard-coded strings that reflect your file system structure. I am not sure if this could be generalised somehow, but yeah, it is a bit of a problem. URLPattern does work for most other scenarios though, but it is tricky.

**Comment by bennettdams** [comment]
I created a feature request for the missing `page` in `v12.2`:  https://github.com/vercel/next.js/discussions/38296
